lib/image-stream/handlers.js

```diff
@@ -123,6 +124,46 @@ module.exports = {
   },
 
   /**
+   * @summary Handle Apple disk images (.dmg)
+   * @function
+   * @public
+   * @memberof handlers
+   *
+   * @param {String} file - file path
+   * @param {Object} options - options
+   * @param {Number} [options.size] - file size
+   *
+   * @fulfil {Object} - image metadata
+   * @returns {Promise}
+   */
+  'application/x-apple-diskimage': (file, options) => {
+    return new Bluebird((resolve, reject) => {
+      const dmg = new UDIF.Image(file);
```

Comment:
Or even better:

```js
const dmg = Bluebird.promisifyAll(new UDIF.Image(file));

return dmg.openAsync().return({
  stream: UDIF.createReadStream(file),
  size: {
    original: options.size,
    final: {
      estimation: false,
      value: dmg.getUncompressedSize()
    }
  },
  transform: new PassThroughStream()
}).tap(dmg.closeAsync);
```

And now I'm going to stop over-optimizing this promise chain :P